File: cpp/ExcelSheetColumnNumber.c
int titleToNumber(char *s) {
    int i = 0, val = 0, x = 1;
    
    if (s[0] == 0) { return 0; }
    for (; s[i+1] != 0; i ++);
    
    for (; i >= 0; i --) {
        val += x * (s[i] - 'A' + 1);
        x *= 26;
    }
    
    return val;
}

File: lc/python/ExcelSheetColumnNumber.py
class Solution:
    # @param s, a string
    # @return an integer
    def titleToNumber(self, s):
        l = len(s)
        x = 1
        val = 0
        
        for ii in xrange(l-1, -1, -1):
            val += x * (ord(s[ii]) - 64)
            x *= 26
        
        return val